sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> The info w.r.t Agent Upgrade. </summary>
+    public partial class AgentUpgrade
+    {
+        /// <summary> Initializes a new instance of AgentUpgrade. </summary>
+        public AgentUpgrade()
+        {
+        }
+
+        /// <summary> Initializes a new instance of AgentUpgrade. </summary>
+        /// <param name="desiredVersion"> Specifies the version info w.r.t AgentUpgrade for the machine. </param>
+        /// <param name="correlationId"> The correlation ID passed in from RSM per upgrade. </param>
+        /// <param name="enableAutomaticUpgrade"> Specifies if RSM should try to upgrade this machine. </param>
+        /// <param name="lastAttemptTimestamp"> Timestamp of last upgrade attempt. </param>
+        /// <param name="lastAttemptStatus"> Specifies the status of Agent Upgrade. </param>
+        /// <param name="lastAttemptMessage"> Failure message of last upgrade attempt if any. </param>
+        internal AgentUpgrade(string desiredVersion, string correlationId, bool? enableAutomaticUpgrade, string lastAttemptTimestamp, LastAttemptStatusEnum? lastAttemptStatus, string lastAttemptMessage)
+        {
+            DesiredVersion = desiredVersion;
+            CorrelationId = correlationId;
+            EnableAutomaticUpgrade = enableAutomaticUpgrade;
+            LastAttemptTimestamp = lastAttemptTimestamp;
+            LastAttemptStatus = lastAttemptStatus;
+            LastAttemptMessage = lastAttemptMessage;
+        }
+
+        /// <summary> Specifies the version info w.r.t AgentUpgrade for the machine. </summary>
+        public string DesiredVersion { get; set; }
+        /// <summary> The correlation ID passed in from RSM per upgrade. </summary>
+        public string CorrelationId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

Reply:
changed the type of CorrelationId to Guid.